sdk/eventhub/azure-eventhub/azure/eventhub/aio/_producer_client_async.py

```diff
@@ -181,11 +344,44 @@ def _create_producer(
         return handler
 
     @classmethod
+    @overload
     def from_connection_string(
         cls,
         conn_str: str,
         *,
         eventhub_name: Optional[str] = None,
+        buffered_mode: Literal[False] = False,
+        **kwargs: Any
+    ) -> "EventHubProducerClient":
+        ...
+
+    @classmethod
+    @overload
+    def from_connection_string(
+        cls,
+        conn_str: str,
+        *,
+        eventhub_name: Optional[str] = None,
+        buffered_mode: Literal[True],
+        on_error: Callable[[SendEventTypes, Optional[str], Exception], Awaitable[None]],
+        on_success: Callable[[SendEventTypes, Optional[str]], Awaitable[None]],
+        max_buffer_length: int = 1500,
+        max_wait_time: float = 1,
+        **kwargs: Any
+    ) -> "EventHubProducerClient":
+        ...
+
+    @classmethod
+    def from_connection_string(
+        cls,
+        conn_str: str,
+        *,
+        eventhub_name: Optional[str] = None,
+        buffered_mode: bool = False,
+        on_error: Optional[Callable[[SendEventTypes, Optional[str], Exception], Awaitable[None]]] = None,
+        on_success: Optional[Callable[[SendEventTypes, Optional[str]], Awaitable[None]]] = None,
+        max_buffer_length: Optional[int] = None,
+        max_wait_time: Optional[float] = None,
```

Comment:
same question about why no typehints for sync implementation.

Reply:
updated, thanks for pointing out!